docs/faq.rst

```diff
@@ -248,6 +248,40 @@ that that were ``malloc()``-ed in another shared library, using data
 structures with incompatible ABIs, and so on. pybind11 is very careful not
 to make these types of mistakes.
 
+How can I properly handle Ctrl-C in long-running functions?
+===========================================================
+
+Ctrl-C is received by the Python interpreter, and holds it until the GIL
+is released, so a long-running function won't be interrupted.
+
+To interrupt from inside your function, you can use the ``PyErr_CheckSignals()`` function,
+that will tell if a signal has been raised on the Python side.
+It's a flag check so its impact is negligible. If a signal is on, interrupt execution by
```

Comment:
"It's [..]" -> "This function merely checks a flag, [..]. When a signal has been received, you can explicitly interrupt"

Reply:
Thanks, updated wording